tests/allheaders.cpp

```diff
@@ -0,0 +1,60 @@
+///////////////////////////////////////////////////////////////////////////////
+// Name:        tests/allheaders.cpp
+// Purpose:     Compilation test for all headers
+// Author:      Arrigo Marchiori
+// Created:     2020-04-20
+// Copyright:   (c) 2010 Vadim Zeitlin, Wlodzimierz Skiba, Arrigo Marchiori
+///////////////////////////////////////////////////////////////////////////////
+
+// ----------------------------------------------------------------------------
+// headers
+// ----------------------------------------------------------------------------
+
+#include "testprec.h"
+
+#ifdef __BORLANDC__
+    #pragma hdrstop
+#endif
+
+#ifndef WX_PRECOMP
+    #include "wx/wx.h"
+#endif // WX_PRECOMP
+
+#include "allheaders.h"
+
+// ----------------------------------------------------------------------------
+// test class
+// ----------------------------------------------------------------------------
+
+class AllHeadersTestCase : public CppUnit::TestCase
+{
+public:
+    AllHeadersTestCase();
+
+private:
+    CPPUNIT_TEST_SUITE(AllHeadersTestCase);
+        CPPUNIT_TEST(AllHeadersTest);
+    CPPUNIT_TEST_SUITE_END();
+
+    void AllHeadersTest();
+
+    wxDECLARE_NO_COPY_CLASS(AllHeadersTestCase);
+};
+
+// register in the unnamed registry so that these tests are run by default
+CPPUNIT_TEST_SUITE_REGISTRATION(AllHeadersTestCase);
+
+// also include in its own registry so that these tests can be run alone
+CPPUNIT_TEST_SUITE_NAMED_REGISTRATION(AllHeadersTestCase,
+                                      "AllHeadersTestCase");
+
+AllHeadersTestCase::AllHeadersTestCase()
+{
+}
+
+void AllHeadersTestCase::AllHeadersTest()
+{
+    /* We compiled, therefore we are successful! */
+    CPPUNIT_ASSERT(true);
+}
+
```

Comment:
All this boilerplate can be replaced with just:
```suggestion
TEST_CASE("wxNO_IMPLICIT_WXSTRING_ENCODING", "[string]")
{
    // Just a dummy test to indicate that we executed -- the main purpose of this test is to check compilation.
    REQUIRE(true);
}
```